src/components/mapgl-layer-icon/mapgl-layer-icon.component.ts

```diff
@@ -0,0 +1,302 @@
+import { Legend, PROPERTY_SELECTOR_SOURCE } from '../mapgl-legend/mapgl-legend.component';
+import { TranslateService } from '@ngx-translate/core';
+import { select } from 'd3-selection';
+
+@Component({
+  selector: 'arlas-mapgl-layer-icon',
+  templateUrl: './mapgl-layer-icon.component.html',
+  styleUrls: ['./mapgl-layer-icon.component.css']
+})
+export class MapglLayerIconComponent implements OnInit, AfterViewInit, OnChanges {
+  @Input() public layer: mapboxgl.Layer;
+  @Input() public colorLegend: Legend = {};
+  @Input() public widthLegend: Legend = {};
+  @Input() public radiusLegend: Legend = {};
+  @ViewChild('layer_icon', { read: ElementRef, static: false }) public layerIconElement: ElementRef;
+
+  constructor(public translate: TranslateService) { }
+
+  public ngOnInit() {
+  }
+
+  public ngAfterViewInit() {
+    if (this.layer) {
+      this.drawIcons();
+    }
+  }
+
+  public ngOnChanges(changes: SimpleChanges) {
+    if (changes['layer'] !== undefined) {
+      if (this.layer) {
+        this.drawIcons();
+      }
+    }
+  }
+
+  private drawIcons(): void {
+    const type = this.layer.type;
+    const paint = this.layer.paint;
+    const source: string = this.layer.source as string;
+    switch (type) {
+      case 'circle': {
+        const p: mapboxgl.CirclePaint = (paint as mapboxgl.CirclePaint);
+        if (source.startsWith('feature') && !source.startsWith('feature-metric')) {
+          drawFeatureCircleIcon(this.layerIconElement.nativeElement, this.colorLegend);
+        } else {
+          // todo
+        }
+        break;
+      }
+      case 'line': {
+        const p: mapboxgl.LinePaint = (paint as mapboxgl.LinePaint);
+        if (source.startsWith('feature') && !source.startsWith('feature-metric')) {
+          drawFeatureLineIcon(this.layerIconElement.nativeElement, this.colorLegend);
+        } else {
+          // todo
+        }
+        break;
+      }
+      case 'fill': {
+        const p: mapboxgl.FillPaint = (paint as mapboxgl.FillPaint);
+        if (source.startsWith('cluster')) {
+          drawClusterFillIcon(this.layerIconElement.nativeElement, this.colorLegend);
+        } else {
+          // todo
+        }
+        break;
+      }
+      case 'heatmap': {
+        const p: mapboxgl.LinePaint = (paint as mapboxgl.LinePaint);
+        if (source.startsWith('cluster')) {
+          drawClusterHeatmapIcon(this.layerIconElement.nativeElement, this.colorLegend);
+        } else {
+          // todo
+        }
+        break;
+      }
+    }
+  }
+
+}
+
+/**
+ * draws the rectangles icon for cluster mode
+ * @param svgNode SVG element on which we append the rectangles using d3.
+ * @param colorLegend Color legend, to give the drawn icons rectagles the same color on the map
+ */
+export function drawClusterFillIcon(svgNode: SVGElement, colorLegend: Legend) {
+  const fourColors = [];
+  if (colorLegend.type === PROPERTY_SELECTOR_SOURCE.interpolated) {
```

Comment:
PROPERTY_SELECTOR_SOURCE is local to arlas-conf, not sure this is the best name here